### PR TITLE
Add basic BGP module tests

### PR DESCRIPTION
**Pull request by marekzatko**
Tests cover if BGP is correctly initialized.
Also fix minor sonar issues.

Signed-off-by: marekzatko <Marek.Zatko@pantheon.tech>

### CODE REVIEW

**Review by PeterSuna — APPROVED** [review]

File: lighty-modules/lighty-bgp/src/test/java/io/lighty/modules/bgp/deployer/TestBase.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    private static final Logger LOG = LoggerFactory.getLogger(TestBase.class);
+    protected static final long WAIT_TIME = 20_000;
+
+    protected static BgpModule bgpModule;
+    protected static LightyServices lightyServices;
+    protected static LightyController controller;
+
+    @BeforeAll
+    static void setup() throws ConfigurationException, InterruptedException, ExecutionException, TimeoutException {
+        ControllerConfiguration controllerConfiguration
```
Comment (vainyksi) [inline lighty-modules/lighty-bgp/src/test/java/io/lighty/modules/bgp/deployer/TestBase.java:39]: make it final

File: lighty-modules/lighty-bgp/src/test/java/io/lighty/modules/bgp/deployer/TestBase.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+import io.lighty.modules.bgp.config.BgpConfigUtils;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+abstract class TestBase {
```
Comment (vainyksi) [inline lighty-modules/lighty-bgp/src/test/java/io/lighty/modules/bgp/deployer/TestBase.java:29]: put all these `@BeforeAll` and `@AfterAll` methods to actual test class `ModuleInitTest`
extract these methods to "Base class" only when it is needed

**Review by vainyksi — CHANGES_REQUESTED** [review]

**Comment by vainyksi** [comment]
Coverage was not calculated for `lighty-examples/lighty-bgp-community-restconf-app/src/main/java/io/lighty/examples/controllers/bgpapp/Main.java`

https://sonarcloud.io/component_measures?branch=bgp-test&id=PantheonTechnologies_lighty-core&metric=new_coverage&selected=PantheonTechnologies_lighty-core%3Alighty-examples%2Flighty-bgp-community-restconf-app%2Fsrc%2Fmain%2Fjava%2Fio%2Flighty%2Fexamples%2Fcontrollers%2Fbgpapp%2FMain.java&view=list

Weird when there is the test `lighty-examples/lighty-bgp-community-restconf-app/src/test/java/io/lighty/examples/controllers/bgpapp/MainTest.java`

File: lighty-modules/lighty-bgp/src/test/java/io/lighty/modules/bgp/deployer/TestBase.java (hunk 1 above)
Comment (marekzatko) [reply lighty-modules/lighty-bgp/src/test/java/io/lighty/modules/bgp/deployer/TestBase.java]: done

File: lighty-modules/lighty-bgp/src/test/java/io/lighty/modules/bgp/deployer/TestBase.java (hunk 2 above)
Comment (marekzatko) [reply lighty-modules/lighty-bgp/src/test/java/io/lighty/modules/bgp/deployer/TestBase.java]: done

Comment (marekzatko) [reply lighty-modules/lighty-bgp/src/test/java/io/lighty/modules/bgp/deployer/TestBase.java]: done

**Review by vainyksi — APPROVED** [review]

**Pull request merged**